lib/modules/dosomething/dosomething_signup/includes/Signup.php

```diff
@@ -91,10 +91,9 @@ private function build($data) {
     $this->campaign = Campaign::get($data->nid);
     $this->campaign_run = $data->run_nid;
 
-    try {
+    if ($data->rbid) {
```

Comment:
this should probably be `if(isset($data->rbid && is_numeric($data->rbid))` to avoid warnings